aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncDeltaSync.java

```diff
@@ -0,0 +1,601 @@
+            Log.v(TAG, "Delta Sync: executing base query from cache");
+        }
+        else {
+            Log.v(TAG, "Delta Sync: executing base query from network");
+        }
+
+        //Execute the base query.
+        awsAppSyncClient.query(baseQuery)
+                .responseFetcher(fetchPolicy)
+                .enqueue(cb);
+
+        try {
+            baseQueryCountdownLatch.await();
+        }
+        catch (InterruptedException iex) {
+            Log.e(TAG, "Delta Sync: Base Query wait failed with [" + iex + "]");
+            deltaSyncOperationFailed = true;
+        }
+
+    }
+
+    void subscribe( ) {
+        Log.v(TAG, "Delta Sync: Subscription was passed in. Setting it up");
+        //Setup an internal callback for the handoff
+        Log.v(TAG, "Delta Sync: Setting mode to QUEUING");
+
+        if (scb == null ) {
+            scb = new AppSyncSubscriptionCall.Callback() {
+                @Override
+                public void onResponse(@Nonnull Response response) {
+                    Log.d(TAG, "Got a Message. Current mode is " + mode);
+                    synchronized (processingLock) {
+                        if (mode == QUEUING_MODE) {
+                            Log.v(TAG, "Delta Sync: Message received while in QUEUING mode. Adding to queue");
+                            messageQueue.add(response);
+                        } else {
+                            Log.v(TAG, "Delta Sync: Message received while in PROCESSING mode.");
+                            lastRunTimeInMilliSeconds = System.currentTimeMillis();
+                            dbHelper.updateLastRunTime(id, lastRunTimeInMilliSeconds);
+                            Log.v(TAG, "Delta Sync: Updating lastRunTime to [" + lastRunTimeInMilliSeconds + "]");
+                            if (subscriptionCallback != null) {
+                                Log.v(TAG, "Delta Sync: Propagating received message");
+                                subscriptionCallback.onResponse(response);
+                            }
+                        }
+                    }
+                }
+
+                @Override
+                public void onFailure(@Nonnull ApolloException e) {
+                    Log.e(TAG, "Delta Sync: onFailure executed with exception: [" + e.getLocalizedMessage() + "]");
+                    //Propagate
+                    if (subscriptionCallback != null) {
+                        Log.v(TAG, "Delta Sync: Propagating onFailure");
+                        subscriptionCallback.onFailure(e);
+                    }
+                }
+
+                @Override
+                public void onCompleted() {
+                    //Resubscribe if there was an error
+                    if (cancelled) {
+                        Log.d(TAG, "Delta Sync [" + id + "]: Received disconnect for subscription. Expected. Will propagate.");
+                        subscriptionCallback.onCompleted();
```

Comment:
On the iOS side, I am not passing this to the developer since we are going to retry and the subscription (sync concept) is never completed. Lets sync on the behavior.

Reply:
Best to stay consistent. Also, with the cancellable object, the developer can get the same info  through the isCancelled method. Will remove.